src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+        }
+    }
+
+    // Multiplier amount, given the length of the lock
+    function lockMultiplier(uint256 secs) public view returns (uint256) {
+        uint256 lock_multiplier = uint256(_MultiplierPrecision) +
+            ((secs * (lockMaxMultiplier - _MultiplierPrecision)) /
+                (lockTimeForMaxMultiplier));
+        if (lock_multiplier > lockMaxMultiplier)
+            lock_multiplier = lockMaxMultiplier;
+        return lock_multiplier;
+    }
+
+    function _averageDecayedLockMultiplier(
+        address account,
+        uint256 index,
+        uint256 elapsedSeconds
+    ) internal view returns (uint256) {
+        return
+            (2 *
+                _lastUsedMultiplier[account][index] -
+                (elapsedSeconds - 1) *
+                multiplierDecayPerSecond) / 2;
+    }
+
+    function setJar(address _jar) external onlyGov {
+        require(_jar != address(0), "cannot set to zero");
+        require(_jar != address(jar), "Jar is already set");
+        jar = IJar(_jar);
+    }
+
+    function setAuthoriseAddress(address _account, bool value)
+        external
+        onlyGov
+    {
+        require(
+            authorisedAddress[_account] != value,
+            "address is already set to given value"
+        );
+        authorisedAddress[_account] = value;
+    }
+
+    function derivedBalance(address account) public returns (uint256) {
+        uint256 _balance = balanceOf(account);
+        uint256 _derived = (_balance * 40) / 100;
+        uint256 _adjusted = (((totalSupply() * DILL.balanceOf(account)) /
+            DILL.totalSupply()) * 60) / 100;
+        uint256 dillBoostedDerivedBal = Math.min(
+            _derived + _adjusted,
+            _balance
+        );
+
+        // Loop through the locked stakes, first by getting the liquidity * lock_multiplier portion
+        uint256 lockBoostedDerivedBal = 0;
+        for (uint256 i = 0; i < _lockedStakes[account].length; i++) {
+            LockedStake memory thisStake = _lockedStakes[account][i];
+            uint256 lock_multiplier = thisStake.lock_multiplier;
+            uint256 lastRewardClaimTime = _lastRewardClaimTime[account];
+            // If the lock is expired
+            if (
+                thisStake.ending_timestamp <= block.timestamp &&
+                !thisStake.isPermanentlyLocked
+            ) {
+                // If the lock expired in the time since the last claim, the weight needs to be proportionately averaged this time
+                if (lastRewardClaimTime < thisStake.ending_timestamp) {
+                    uint256 timeBeforeExpiry = thisStake.ending_timestamp -
+                        lastRewardClaimTime;
+                    uint256 timeAfterExpiry = block.timestamp -
+                        thisStake.ending_timestamp;
+
+                    // Get the weighted-average lock_multiplier
+                    uint256 numerator = (lock_multiplier * timeBeforeExpiry) +
```

Comment:
As I understand it, claims that are made after the lock expires receive the original undecayed lock. For example, if a user locks for 1 year and doesn't make a claim until 1 year and 1 days later, they will receive a full 2.5x boost for their lock (when it would have normally decayed if they claimed throughout the year).

Reply:
Yes I will average the decay.